smith-carson/hiring-project
# Kevin boucher

Pull request for Kevin Boucher's solution to the Smith & Carson hiring code exercise.

**Demo**
https://user-images.githubusercontent.com/1943568/225469541-5f11aae8-d5ba-473c-aa37-d55e0efa33cc.mov

* Upgrade Ember.js version to 3.28.4 (not upgrading to v4 in case IE 11 support is still required)
* Run linters with `--fix` option (would like to change some of these rules)
* Add SASS support via `ember-cli-sass` addon
* Add `ember-bootstrap` addon
* Upgrade `ember-auto-import` to support `ember-bootstrap` addon
* Update router to make `candidates` the default route
* Import Smith & Carson fonts from Google fonts API
* Add base styles and layout elements and hide _Candidates_ link
* Add `ember-intl` addon for translations, update raw text in templates, convert candidates list to table and style _Add new_ button
* Update prettier tab size config (and apply)
* Add `ember-truth-helpers` addon
* Add Candidate route and form component
    * Add `CandidatesCandidateRoute`
    * Add `CandidatesCandidateController`
    * Add `ApplicantFormComponent`
    * Add Bootstrap `form-control` style overrides for invalid field treatment
    * Add `ember-test-selectors` addon
    * Remove `CandidatesCandidateController#addNew` action in favor of `LinkTo` detail route
    * Remove `outlet` from list view template
* Ensure candidates list is refreshed after adding/updating a candidate
* Update README.md structure
* Add whitespace to the bottom of the main layout container
* Add POST route to Phalcon server index
* Standardize invalid form field treatment (leverage Bootstrap `is-invalid`)
